File: apps/api/app/schemas/calls.py
from datetime import datetime
from typing import Literal
from uuid import UUID

from pydantic import BaseModel, Field

class CallResponse(BaseModel):
    id: UUID
    status: str
    caller_number: str | None
    started_at: datetime | None
    ended_at: datetime | None
    duration_seconds: int | None
    minutes_charged: int | None
    summary_text: str | None
    recording_url: str | None


class CallSummaryResponseFields(BaseModel):
    summary_status: Literal["processing", "ready", "unavailable"]
    caller_intent: str | None
    action_items: list[str] | None
    sentiment: str | None
    follow_up_required: bool | None


class CallHistoryListItem(CallSummaryResponseFields):
    id: UUID
    status: str
    caller_number: str | None
    started_at: datetime | None
    ended_at: datetime | None
    duration_seconds: int | None
    minutes_charged: int | None
    summary_text: str | None
    has_recording: bool


class CallHistoryListResponse(BaseModel):
    calls: list[CallHistoryListItem]
    total: int = Field(ge=0)
    limit: int = Field(ge=1, le=100)
    offset: int = Field(ge=0)
    has_more: bool


class CallTranscriptLineResponse(BaseModel):
    speaker: str
    text: str
    sequence_number: int
    created_at: datetime


class CallDetailResponse(CallSummaryResponseFields):
    id: UUID
    status: str
    caller_number: str | None
    started_at: datetime | None
    ended_at: datetime | None
    duration_seconds: int | None
    minutes_charged: int | None
    summary_text: str | None
    recording_url: str | None
    transcript: list[CallTranscriptLineResponse]

File: apps/web/src/lib/types/calls.ts
export type CallSummaryStatus = "processing" | "ready" | "unavailable";

export type CallSummaryFields = {
  summary_status: CallSummaryStatus;
  caller_intent: string | null;
  action_items: string[] | null;
  sentiment: string | null;
  follow_up_required: boolean | null;
};

export type CallHistoryListItem = CallSummaryFields & {
  id: string;
  status: string;
  caller_number: string | null;
  started_at: string | null;
  ended_at: string | null;
  duration_seconds: number | null;
  minutes_charged: number | null;
  summary_text: string | null;
  has_recording: boolean;
};

export type CallHistoryListResponse = {
  calls: CallHistoryListItem[];
  total: number;
  limit: number;
  offset: number;
  has_more: boolean;
};

export type CallTranscriptLine = {
  speaker: string;
  text: string;
  sequence_number: number;
  created_at: string;
};

export type CallDetail = CallSummaryFields & {
  id: string;
  status: string;
  caller_number: string | null;
  started_at: string | null;
  ended_at: string | null;
  duration_seconds: number | null;
  minutes_charged: number | null;
  summary_text: string | null;
  recording_url: string | null;
  transcript: CallTranscriptLine[];
};
